File: src/main/java/sunshine/api/weather/WeatherController.java
package sunshine.api.weather;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import sunshine.api.common.ApiResponse;
import sunshine.application.weather.WeatherDto;
import sunshine.application.weather.WeatherFacade;

@RestController
@RequestMapping("/api/v1/weather")
public class WeatherController {

    private final WeatherFacade weatherFacade;

    public WeatherController(WeatherFacade weatherFacade) {
        this.weatherFacade = weatherFacade;
    }

    @GetMapping
    public ApiResponse<WeatherDto> getWeather(@RequestParam String city) {
        if (city == null || city.isBlank()) {
            throw new IllegalArgumentException("City name is required");
        }
        return ApiResponse.ok(weatherFacade.getWeather(city));
    }
}

File: src/main/kotlin/com/sunshine/api/controller/WeatherController.kt
package com.sunshine.api.controller

import com.sunshine.api.dto.WeatherResponse
import com.sunshine.api.response.ApiResponse
import com.sunshine.application.WeatherFacade
import jakarta.validation.constraints.NotBlank
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/weather")
@Validated
class WeatherController(
    private val weatherFacade: WeatherFacade
) {
    @GetMapping
    fun getWeather(@RequestParam("city") @NotBlank(message = "도시 이름을 입력하세요.") city: String): ApiResponse<WeatherResponse> {
        val response = weatherFacade.getWeather(city)
        return ApiResponse.success(response)
    }
}
